src/core/Trigger.helper.spec.js

```diff
@@ -0,0 +1,53 @@
+import Sinon from 'sinon';
+import TriggerHelper from './Trigger.helper';
+import Producer from './Producer';
+
+describe('TriggerHelper Unit Tests', () => {
+
+  it('should trigger the action within the expected time frame', async () => {
+
+    const ProducerMock = Sinon.mock(Producer);
+    ProducerMock.collect = Sinon.spy();
+
+    const delayedCondition = async function () {
+      const x = Math.floor(Math.random() * 6) + 1;
+      return x === 6;
```

Comment:
@robertdumitrescu INFO - Technically, this could be a flaky test if you were unlucky enough not to get a 6.

Reply:
Pf...Yeah...but what is the chance since is running 100 times? I wanted to see that is run at least 1 time and without having a super convoluted way of doing it.

Reply:
I can raise the chance to 33% by doing 0-3 interval if that alleviates your concerns.